nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/test/java/org/apache/nifi/processors/gcp/pubsub/PublishGCPubSubTest.java

```diff
@@ -0,0 +1,230 @@
+import static org.mockito.Mockito.when;
+
+public class PublishGCPubSubTest {
+
+    private Throwable throwable;
+    private Publisher publisherMock;
+    private TestRunner runner;
+
+    @BeforeEach
+    void setRunner() {
+        throwable = null;
+        publisherMock = mock(Publisher.class);
+        runner = TestRunners.newTestRunner(new PublishGCPubSub() {
+            @Override
+            @OnScheduled
+            public void onScheduled(ProcessContext context) {
+                publisher = publisherMock;
+            }
+
+            @Override
+            protected void addCallback(ApiFuture<String> apiFuture, ApiFutureCallback<? super String> callback, Executor executor) {
+                if (callback instanceof NiFiApiFutureCallback) {
+                    final NiFiApiFutureCallback niFiApiFutureCallback = (NiFiApiFutureCallback) callback;
+                    if (throwable == null) {
+                        niFiApiFutureCallback.onSuccess(Long.toString(System.currentTimeMillis()));
+                    } else {
+                        niFiApiFutureCallback.onFailure(throwable);
+                    }
+                }
+            }
+        });
+    }
+
+    @Test
+    void testPropertyDescriptors() throws InitializationException {
+        runner.assertNotValid();
+
+        runner.setProperty(PublishGCPubSub.GCP_CREDENTIALS_PROVIDER_SERVICE, getCredentialsServiceId(runner));
+        runner.assertNotValid();
+
+        runner.setProperty(PublishGCPubSub.TOPIC_NAME, "my-topic");
+        runner.assertNotValid();
+
+        runner.setProperty(PublishGCPubSub.PROJECT_ID, "my-project");
+        runner.assertValid();
+
+        runner.setProperty(PublishGCPubSub.API_ENDPOINT, "localhost");
+        runner.assertNotValid();
+        runner.setProperty(PublishGCPubSub.API_ENDPOINT, "localhost:443");
+        runner.assertValid();
+
+        runner.setProperty(PublishGCPubSub.BATCH_SIZE_THRESHOLD, "-1");
+        runner.assertNotValid();
+        runner.setProperty(PublishGCPubSub.BATCH_SIZE_THRESHOLD, "15");
+        runner.assertValid();
+
+        runner.setProperty(PublishGCPubSub.BATCH_BYTES_THRESHOLD, "3");
+        runner.assertNotValid();
+        runner.setProperty(PublishGCPubSub.BATCH_BYTES_THRESHOLD, "3 MB");
+        runner.assertValid();
+
+        runner.setProperty(PublishGCPubSub.BATCH_DELAY_THRESHOLD, "100");
+        runner.assertNotValid();
+        runner.setProperty(PublishGCPubSub.BATCH_DELAY_THRESHOLD, "100 millis");
+        runner.assertValid();
+    }
+
+    @Test
+    void testSendOne_Success_FlowFileStrategy() throws InitializationException {
+        runner.setProperty(PublishGCPubSub.GCP_CREDENTIALS_PROVIDER_SERVICE, getCredentialsServiceId(runner));
+        runner.setProperty(PublishGCPubSub.TOPIC_NAME, "my-topic");
+        runner.setProperty(PublishGCPubSub.PROJECT_ID, "my-project");
+
+        runner.enqueue("text");
+        runner.run(1, true, true);
```

Comment:
This is the same as `TestRunner.run()`, so recommend removing the arguments and using the default behavior.
```suggestion
        runner.run();
```